accepted/2024/package-update-command.md

```diff
@@ -0,0 +1,333 @@
+ Updated Newtonsoft.Json 11.0.1 to 13.0.3
+    
+Updated 2 packages in 36 scanned packages.
+```
+
+### Fixing vulnerabilities
+
+When users run `dotnet build` or `dotnet restore` commands in CLI, if they see any warnings related to vulnerabilities in their project’s NuGet dependencies, they can run `dotnet package update --vulnerable` CLI command to fix all the mentioned issues.
+
+In the future this can be extended to fix deprecated versions.
+
+## Explanation
+
+### Functional explanation
+
+<!-- Explain the proposal as if it were already implemented and you're teaching it to another person. -->
+<!-- Introduce new concepts, functional designs with real life examples, and low-fidelity mockups or  pseudocode to show how this proposal would look. -->
+
+#### dotnet audit
+
+The `dotnet audit` command will scan your project and solution's dependencies and look for known vulnerabilities, deprecations, and outdated packages. If any vulnerabilities, deprecations, or outdated packages are found, a remediation will be calculated to then `fix` which will be applied to the resulting package graph.
+
+Example:
+
+![dotnet audit](/meta/resources/dotnetaudit/dotnetaudit.png)
+
+```
+$ dotnet audit
+
+Fetching package metadata from: 'https://api.nuget.org/v3/index.json'
+Loaded 23 security advisories from 'https://api.nuget.org/v3/index.json'
+Scanning ContosoUniversity.sln (36 NuGet dependencies)
+
+error: Vulnernable packages found!
+[net5.0]:
+Top-level Package     Requested Resolved Severity Advisory URL
+> UmbracoForms      8.4.1  8.4.1  Moderate https://github.com/advisories/GHSA-8m73-w2r2-6xxj
+
+Transitive Package  Resolved Severity Advisory URL
+> Microsoft.Data.OData 5.2.0  Moderate https://github.com/advisories/GHSA-mv2r-q4g5-j8q5
+
+Found 1 top-level Moderate severity vulnerability & 1 transtive Moderate severity vulnerability package(s) in 36 scanned packages.
+
+Run 'dotnet audit fix' to fix them.
+
+warning: Deprecated packages found!
+[net5.0]:
+Top-level Package      Requested Resolved Reason(s) Alternative
+> anurse.testing.TestDeprecatedPackage 1.0.0  1.0.0  Legacy  Microsoft.AspNetCore.Mvc > 0.0.0
+
+Found 1 top-level Legacy deprecated package(s) in 36 scanned packages.
+
+Run 'dotnet audit fix' to fix them.
+
+warning: Outdated packages found!
+[net5.0]:
+Top-level Package                  Resolved Latest
+> anurse.testing.TestDeprecatedPackage             1.0.0  2.0.0
+> UmbracoForms                   8.4.1     8.7.1
+
+Found 2 top-level outdated package(s) in 36 scanned packages.
+
+Run 'dotnet audit fix' to fix them.
+```
+
+##### Endpoints
+
+NuGet will use existing endpoints to optimize the speed of audit results.
+
+- [Deprecation](https://docs.microsoft.com/en-us/nuget/api/registration-base-url-resource#package-deprecation)
+- [Vulnerability](https://docs.microsoft.com/en-us/nuget/api/registration-base-url-resource#vulnerabilities)
+- Outdated - No existing endpoint, will need to call a source.
```

Comment:
The way outdated packages are also determined from the registration endpoint. 

It has a list of all available versions, listed/unlisted, so it can just determine from that.